crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -12,24 +12,53 @@ pub mod contracts;
 
 pub use self::contracts::Contracts;
 
+/// An Ethereum RPC connection.
+pub struct Rpc {
+    web3: DynWeb3,
+    network: Network,
+}
+
+/// Network information for an Ethereum blockchain connection.
+#[derive(Clone, Debug)]
+pub struct Network {
+    pub id: eth::NetworkId,
+    pub chain: eth::ChainId,
+}
+
+impl Rpc {
+    /// Instantiate an RPC connection to an Ethereum (or Ethereum-compatible)
+    /// node at the specifed URL.
+    pub async fn connect(url: &url::Url) -> Result<Self, Error> {
+        let web3 = boundary::buffered_web3_client(url);
+        let id = web3.net().version().await?.into();
+        let chain = web3.eth().chain_id().await?.into();
+
+        Ok(Self {
+            web3,
+            network: Network { id, chain },
+        })
+    }
+
+    /// Returns the network information for the RPC connection.
+    pub fn network(&self) -> &Network {
+        &self.network
+    }
+}
+
 /// The Ethereum blockchain.
 #[derive(Clone)]
 pub struct Ethereum {
     web3: DynWeb3,
-    chain_id: eth::ChainId,
-    network_id: eth::NetworkId,
+    network: Network,
     contracts: Contracts,
     gas: Arc<NativeGasEstimator>,
 }
 
 impl Ethereum {
-    /// Access the Ethereum blockchain through an RPC API hosted at the given
-    /// URL.
-    pub async fn ethrpc(url: &url::Url, addresses: contracts::Addresses) -> Result<Self, Error> {
-        let web3 = boundary::buffered_web3_client(url);
-        let chain_id = web3.eth().chain_id().await?.into();
-        let network_id = web3.net().version().await?.into();
-        let contracts = Contracts::new(&web3, &network_id, addresses);
+    /// Access the Ethereum blockchain through an RPC API.
+    pub async fn new(rpc: Rpc, addresses: contracts::Addresses) -> Result<Self, Error> {
```

Comment:
side note: Since this type is essential for running the driver we might as well panic on an error here and make error handling `Etherum::new()` obsolete.